backend/controllers/user.js

```diff
@@ -1,48 +1,111 @@
+    const { loggedUser } = req
+    if (!loggedUser) throw new UnauthorizedError()
 
-    loggedUser.dataValues.email = req.headers.email;
-    delete req.headers.email;
+    loggedUser.dataValues.email = req.headers.email
+    delete req.headers.email
 
-    res.json({ user: loggedUser });
+    res.json({ user: loggedUser })
   } catch (error) {
-    next(error);
+    next(error)
   }
-};
+}
 
 //* Update User
 const updateUser = async (req, res, next) => {
   try {
-    const { loggedUser } = req;
-    if (!loggedUser) throw new UnauthorizedError();
+    const { loggedUser } = req
+    if (!loggedUser) throw new UnauthorizedError()
 
     const {
       user: { password },
       user,
-    } = req.body;
+    } = req.body
 
     Object.entries(user).forEach((entry) => {
-      const [key, value] = entry;
+      const [key, value] = entry
 
-      if (value !== undefined && key !== "password") loggedUser[key] = value;
-    });
+      if (value !== undefined && key !== "password") loggedUser[key] = value
+    })
 
     if (password !== undefined || password !== "") {
-      loggedUser.password = await bcryptHash(password);
+      loggedUser.password = await bcryptHash(password)
     }
 
-    await loggedUser.save();
+    await loggedUser.save()
 
-    res.json({ user: loggedUser });
+    res.json({ user: loggedUser })
   } catch (error) {
-    next(error);
+    next(error)
   }
-};
+}
 
-module.exports = { currentUser, updateUser };
+//* Request Password Reset
+const requestPasswordReset = async (req, res, next) => {
+  try {
+    const { email } = req.body
+    const user = await User.findOne({ where: { email } })
+    if (!user) throw new NotFoundError("User")
+
+    const token = crypto.randomBytes(20).toString('hex')
+    user.resetPasswordToken = token
+    user.resetPasswordExpires = Date.now() + 3600000 // 1 hour
+    await user.save()
+
+    const transporter = nodemailer.createTransport({
+      service: 'Gmail',
+      auth: {
+        user: process.env.EMAIL_ADDRESS,
+        pass: process.env.EMAIL_PASSWORD,
+      },
+    })
+
+    const mailOptions = {
+      to: user.email,
+      from: process.env.EMAIL_ADDRESS,
+      subject: 'Password Reset',
+      text: `You are receiving this because you (or someone else) have requested the reset of the password for your account.\n\n
+             Please click on the following link, or paste this into your browser to complete the process:\n\n
+             http://${req.headers.host}/reset/${token}\n\n
+             If you did not request this, please ignore this email and your password will remain unchanged.\n`,
+    }
+
+    await transporter.sendMail(mailOptions)
+    res.status(200).json({ message: 'Password reset email sent' })
+  } catch (error) {
+    next(error)
+  }
+}
+
+//* Reset Password
+const resetPassword = async (req, res, next) => {
+  try {
+    const { token, newPassword } = req.body
+    const user = await User.findOne({
```

Comment:
The error message 'Password reset token is invalid or has expired.' should be handled using a custom error class for better error management and consistency across the application.